lib/pure/strmisc.nim

```diff
@@ -85,6 +86,84 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: openArray[char]; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - `str` must not be empty.
+  ## - `str` must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - No separator before a digit.
+  ## - First separator can be anywhere after first digit, but no more than 3 characters.
```

Comment:
i was confused before, and I'm still confused now that you put this line back; what does it mean?
first is vague (leftmost or rightmost?)
that would kind of imply one of those 2 is valid (depending on leftmost or rightmost) but that's not the case:

parseFloatThousandSep("10000,00.0")
parseFloatThousandSep("10,00000.0")

Reply:
Ultimately user can just look at the code anyways... :shrug:

Reply:
@timotheecour It means that a *thousand* separator can appear anywhere after the first digit (referring to one, ten, one hundred thousands) but of course no more than 3 characters, because at that point it wouldn't be a thousand separator anymore. 
Of course once the first thousand separator has appeared there *has* to be 3 characters between each thousand separator and between the last and a possible decimal dot.